Cleanup whitelisting in tests/samples

nashif opened the pull request:
Remove whitelisting and expand test coverage.

dbkinder reviewed (commented):
No docs to review (yet)?

nashif committed:
samples: spi_flash: remove whitelisting

Instead of whitelisting, use filtering on configured SPI Flash device.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: grove_temperature: cleanup whitelisting

Do not whitelist on boards, rely on features instead. Turns out we do
not need I2C enabled as well.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
soc: ke1xf: fix SPI device numbering

In previous patch used the wrong index for the SPI devices. There is 0
and 1 instead of 1 and 2.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: build_all: remove whitelisting on arduino_101

Whitelisting was not needed, we can build this for other boards.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: quark_se_c1000_devboard does not have Wifi

Remove custom configuration that does not support Wifi natively.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: hifive1: add aliases for pwm

Add missing aliases and make this board available for some PWM samples.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: hifive1: add supported peripherals

Add supported features to the board definition.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: net: quark/arduino 101 do not have ethernet support

Remove old configurations of boards that do not support networking
natively and very custom and difficult to test.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: arduino_due: this board does support gpio

Add gpio as being supported by this board.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: add supported devices to boards

add supported features (i2c) to boards definitions.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
build: dts: also look for application specific dts overlay

Not all DTS changes are board specific and we should support overlays
that can be used with multiple boards. For example with a sensor, for
example:

Use this overlay for a sensor:

	&arduino_i2c {
		bme280@76 {
			compatible = "bosch,bme280";
			reg = <0x76>;
			label = "BME280";
		};
	};

This will work with any board that has &arduino_i2c defined, but should
also work if you specify the instance directly (&i2c0 or &i2c1).

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
arduino_due: do not build gpio_sam driver

This board uses the sam3 gpio driver. gpio_sam was selected by default
causing a conflict and issues when building for this board.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: mass_storage: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: console: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: mark boards as supporting usb_cdc in DT

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: hid-mouse: use DT filtering instead of platform_exclude

Do not platform_exclude, instead use DT filter.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: hci_usb: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: bluetooth: shell: remove whitelisting on arduino_101

Not a primary bluetooth testing platform.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: i2c_fujitsu_fram: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: pwm: sifive: fix DTS defines

Remove deprecated macros and use DT_INST_* variants instead.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: blink_led: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: fade_led: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: servo_motor: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
drivers: pwm: imx: replace deprecated DTS macros

Replace deprecated DT macros with DT_INST_ variants.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: colibri_imx7d_m4: this board supports PWM

Add pwm as supported feature to the board yaml file.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: pwm_api: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.
Add pwm to board yaml where it applies.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: gpio: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: i2c: remove whitelisting

Remove whitelisting and make the test more generic. This test was only
building on 1 board basically, now with the new filters it builds on 24.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
tests: pinmux: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: webusb: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
samples: led_apa102c_bitbang: remove whitelisting

Remove whitelisting and enable broader testing on all boards with needed
features.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

SebastianBoe commented on cmake/extensions.cmake:
```diff
@@ -346,6 +346,7 @@ endmacro()
 macro(zephyr_library)
   zephyr_library_get_current_dir_lib_name(lib_name)
   zephyr_library_named(${lib_name})
+  zephyr_library_sources(${ZEPHYR_BASE}/misc/empty_file.c)
```
Triggering a build failure when a library has no sources is not a bug, it is a feature.

Removing this feature will cause problems.

For instance, users use the build log to see what is included in their build, if they believe they have
disabled the UART driver, but still see an uart lib being linked in then this will cause confusion.

We should not make things more difficult for our users in the name of making our own testing easier.

nashif replied:
The vanilla behavior of cmake when a cmakelists.txt is empty is to continue and not fail. This is what users expect. This issue is a by-product of how zephyr macros operate, not an intended feature. 

> We should not make things more difficult for our users in the name of making our own testing easier.

we are making it more difficult by changing to cmake to work differently than the default behavior.

SebastianBoe replied:
I don't understand what you mean by a cmakelists.txt being empty.

The default behaviour of CMake is to error-out at generation-time when a library does not have any sources.

> CMake Error at CMakeLists.txt:21 (add_library):
>   No SOURCES given to target: foo
>

SebastianBoe replied:
If you could help me by describing more precisely the problem being solved, preferably with an example, then maybe we can figure something out.

Right now I am just seeing a change, a vague problem description, and a problem that is created by this change.

nashif replied:
forgot about the library requiring at least one source, let me see what other options are available.

nashif replied:
remove specific commit for now.

finikorg commented on samples/subsys/usb/webusb/sample.yaml:
```diff
@@ -3,5 +3,5 @@ sample:
     depends_on: usb_device
-    platform_whitelist: quark_se_c1000_devboard
     tags: usb
+    platform_exclude: native_posix
```
After #17397 is merged this is not needed

nashif replied:
#17397 not helping, still the same issue.

finikorg reviewed (approved):
LGTM with minor comments

MaureenHelm commented on soc/arm/nxp_kinetis/ke1xf/dts_fixup.h:
```diff
@@ -81,19 +81,18 @@
+#define DT_SPI_1_IRQ			DT_NXP_IMX_LPSPI_4002D000_IRQ_0
+#define DT_SPI_1_IRQ_PRI		DT_NXP_IMX_LPSPI_4002D000_IRQ_0_PRIORITY
+#define DT_SPI_1_CLOCK_NAME		DT_NXP_IMX_LPSPI_4002D000_CLOCK_CONTROLLER
+#define DT_SPI_1_CLOCK_SUBSYS		DT_NXP_IMX_LPSPI_4002D000_CLOCK_NAME
```
@henrikbrixandersen fyi

henrikbrixandersen replied:
@MaureenHelm Thanks. The current ones, introduced in ac2b8eee4ac6bd121e5446b7261bd1ac42739be2, are incorrect.

MaureenHelm commented on samples/subsys/usb/console/arduino_101.overlay:
```diff
@@ -1,10 +0,0 @@
- */
-/ {
-        chosen {
-                zephyr,console = &usb_cdc;
```
Do you need to add an app.overlay to set this?

nashif replied:
yep, forgot to 'git add' the file.

MaureenHelm commented on tests/subsys/fs/fat_fs_api/testcase.yaml:
```diff
@@ -1,4 +1,4 @@
 tests:
   filesystem.fat:
-    platform_whitelist: arduino_101 native_posix
+    platform_whitelist: native_posix
```
The commit message says you're enabling "broader testing on all boards with needed
features", so why are you still whitelisting?

nashif replied:
yeah, we going through commits adding bodies almost like a robot. Will change.

nashif replied:
removed this commit from the series for now.

MaureenHelm commented on tests/drivers/pwm/pwm_api/testcase.yaml:
```diff
@@ -1,7 +1,5 @@
     tags: drivers pwm
-    depends_on: pwm
+    filter: DT_ALIAS_PWM_0_LABEL or DT_ALIAS_PWM_1_LABEL or DT_ALIAS_PWM_2_LABEL or DT_ALIAS_PWM_3_LABEL
+    #depends_on: pwm
```
Did you mean to comment this out?

nashif replied:
will be fixed, the yaml data is less complete than what we are getting with the filter, so was trying to get more coverage. adding pwm to all those remaining boards now...

jukkar reviewed (approved):
+1 to networking changes

jfischer-phytec-iot commented on samples/basic/blink_led/src/main.c:
```diff
@@ -15,28 +15,7 @@
-#define PWM_DRIVER	PWM_1_LABEL
-#define PWM_CHANNEL	0
-#elif defined(DT_ALIAS_PWM_LED0_PWMS_CONTROLLER) && defined(DT_ALIAS_PWM_LED0_PWMS_CHANNEL)
+#if defined(DT_ALIAS_PWM_LED0_PWMS_CONTROLLER) && defined(DT_ALIAS_PWM_LED0_PWMS_CHANNEL)
```
Thanks
